bundles/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/discovery/NeatoAccountDiscoveryService.java

```diff
@@ -88,17 +87,17 @@ private void addThing(Robot robot) {
             return;
         }
 
-        logger.debug("addThing(): Adding new Neato unit {} to the inbox", robot.getName());
+        logger.debug("addThing(): Adding new Neato unit ({}) to the inbox", robot.getName());
 
         Map<String, Object> properties = new HashMap<>();
         String serial = robot.getSerial();
-        ThingUID thingUID = new ThingUID(NeatoBindingConstants.THING_TYPE_VACUUMCLEANER, bridgeUID, robot.getSerial());
```

Comment:
This doesn't look right. The `bridgeUID` should be part of the `ThingUID`. So it should not be removed.. What seems to be wrong here is that the `bridgeUID` is not set. This can be done in the constructor by getting the UID from the passed `NeatoAccountHandler`.

Reply:
Thanks.  I originally tried that route but was getting a null uid returned... I just realized that I had tried to get the bridgeUID instead of the thingUID from the handler (which is the bridge).